compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1260,7 +1260,15 @@ OMR::ResolvedMethodSymbol::genIL(TR_FrontEnd * fe, TR::Compilation * comp, TR::S
 
          if (optimizer)
             {
-            optimizer->optimize();
+            try
+               {
+               optimizer->optimize();
+               }
+            catch (const TR::CompilationException &e)
+               {
+               bool success = comp->fe()->handleExceptionThrownFromILGen();
+               _methodFlags.set(IlGenSuccess, success);
```

Comment:
Peeking does not check this flag, nor does inlining. Confirmed with Marius, no one cares about this flag. Maybe we should clean up this flag.